source/declarations.tex

```diff
@@ -1010,7 +1010,10 @@
 \indextext{inline function}%
 A function declaration\iref{dcl.fct,class.mfct,class.friend}
 with an \keyword{inline} specifier declares an
-\defnadj{inline}{function}. The inline specifier indicates to
+\defnadj{inline}{function}.
+
+\recommended
+The \keyword{inline} specifier indicates to
 the implementation that inline substitution of the function body at the
 point of call is to be preferred to the usual function call mechanism.
 An implementation is not required to perform this inline substitution at
```

Comment:
Two lines down is a "shall" which can't be inside "recommended practice".

I agree with the general approach, but this needs more massaging.

Reply:
I've thought about how to rephrase the latter half of the recommendation, and quite frankly, it should just be deleted. Everything stated in the second half is redundant, especially if this is located in a recommended practice paragraph.

The wording already suggests that this inline substitution is not mandatory. On top of that, with this change, the recommendation is in a Recommended Practice paragraph. It is crystal clear that an implementation isn't required to perform this substitution.

Furthermore, it is nowhere stated that not following this recommended practice somehow disables the other rules that apply to inline functions. There is no reason why the reader may assume so, and explicitly stating it is noise.